Xamarin.Forms.Platform.Android/AppCompat/FormsAppCompatActivity.cs

```diff
@@ -494,13 +538,15 @@ public override void OnWindowAttributesChanged(WindowManagerLayoutParams @params
 
 			_isFullScreen = !_isFullScreen;
 
+			_statusBarUnderlay.Visibility = _isFullScreen ? ViewStates.Gone : ViewStates.Visible;
+
 			var displayMetrics = Resources.DisplayMetrics;
 			var width = displayMetrics.WidthPixels;
 			var height = displayMetrics.HeightPixels;
 			AppCompat.Platform.LayoutRootPage(this, Xamarin.Forms.Application.Current.MainPage, width, height);
 		}
 
-		void SetStatusBarVisibility(SoftInput mode)
+		void SetSystemUiVisibility(SoftInput mode)
```

Comment:
Renaming method in order to avoid confusion. Changing UI visibility flags doesn't necessarily change status bar visibility especially in this implementation.